subworkflows/local/longread_hostremoval.nf

```diff
@@ -39,9 +41,22 @@ workflow LONGREAD_HOSTREMOVAL {
     SAMTOOLS_BAM2FQ ( SAMTOOLS_VIEW.out.bam, false )
     ch_versions      = ch_versions.mix( SAMTOOLS_BAM2FQ.out.versions.first() )
 
+    SAMTOOLS_INDEX ( SAMTOOLS_VIEW.out.bam )
+    ch_versions      = ch_versions.mix( SAMTOOLS_INDEX.out.versions.first() )
+
+    SAMTOOLS_VIEW.out.bam
+        .join(SAMTOOLS_INDEX.out.bai, by: [0], remainder: true)
+        .set { bam_bai }
```

Comment:
I personally prefer using assignment rather than `set` because I think it's easier to see where a new channel gets created when scanning code. Also, `by: 0` is the default so you can skip it, I think?

```suggestion
    bam_bai = SAMTOOLS_VIEW.out.bam
        .join(SAMTOOLS_INDEX.out.bai, remainder: true)
```

Reply:
Thanks for the suggestion @Midnighter